JAX v0.3.18 contains breaking change that fails CI

https://github.com/google/jax/releases/tag/jax-v0.3.18 has a breaking change of

> `jax._src` is no longer imported into the from the public `jax` namespace. This may break users that were using JAX internals.

This breaks the current CI with

```
=================================== FAILURES ===================================
________________________ test_diffable_backend_failure _________________________

    def test_diffable_backend_failure():
        pyhf.set_backend("numpy", default=True)
        pyhf.set_backend("jax")
    
        def example_op(x):
            y = pyhf.default_backend.astensor(x)
            return 2 * y
    
        with pytest.raises(
            (
                ValueError,
>               jax._src.errors.TracerArrayConversionError,
                jax.errors.ConcretizationTypeError,
            )
        ):
E       AttributeError: module 'jax' has no attribute '_src'

example_op = <function test_diffable_backend_failure.<locals>.example_op at 0x7f352304d2d0>

tests/test_backends.py:65: AttributeError
```

as

https://github.com/scikit-hep/pyhf/blob/37abb3b13b75fb161ace06535f3204c423191d18/tests/test_backends.py#L62-L77

--- a/tests/test_backends.py
+++ b/tests/test_backends.py
@@ -62,7 +62,7 @@ def test_diffable_backend_failure():
     with pytest.raises(
         (
             ValueError,
-            jax._src.errors.TracerArrayConversionError,
+            jax.errors.TracerArrayConversionError,
             jax.errors.ConcretizationTypeError,
         )
     ):
@@ -73,7 +73,7 @@ def test_diffable_backend_failure():
         z = pyhf.tensorlib.sum(y)
         return z
 
-    with pytest.raises(jax._src.errors.TracerArrayConversionError):
+    with pytest.raises(jax.errors.TracerArrayConversionError):
         jax.jacrev(example_op2)(pyhf.tensorlib.astensor([2.0, 3.0]))
 
 

